tools/cldr-code/src/test/java/org/unicode/cldr/util/TestMatchValue.java

```diff
@@ -0,0 +1,60 @@
+package org.unicode.cldr.util;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.common.collect.ImmutableSet;
+import org.junit.jupiter.api.Test;
+import org.unicode.cldr.util.MatchValue.SemverMatchValue;
+import org.unicode.cldr.util.MatchValue.ValidityMatchValue;
+
+public class TestMatchValue {
+    @Test
+    void TestSemverMatchValue() {
+        // assert that no keyword is allowed
+        assertThrows(
+                IllegalArgumentException.class,
+                () -> {
+                    SemverMatchValue.of("Not Allowed");
+                });
+
+        final SemverMatchValue m = SemverMatchValue.of(null);
+
+        ImmutableSet<String> is_semver = ImmutableSet.of("0.0.0", "1.0.0", "42.0.0-BETA3");
+        assertAll(
+                "is=true",
+                is_semver.stream().map(v -> () -> assertTrue(m.is(v), v + ": Should be a semver")));
+
+        ImmutableSet<String> not_semver = ImmutableSet.of("0.0", "v1.0", "Some Version", "");
+        assertAll(
+                "is=true",
+                not_semver.stream()
+                        .map(v -> () -> assertFalse(m.is(v), v + ": Should NOT be a semver")));
```

Comment:
this works, but forEach should be used where you aren't using the results of the match.

Reply:
actually the map is to an assertion (an Executable), and they are all rolled up into the `assertAll` call.  This is very handy, because if two out of five fail, it will say so rather than stopping at the first failure.

Reply:
```
org.opentest4j.MultipleFailuresError: is=true (2 failures)
        org.opentest4j.AssertionFailedError: mt-Latn: Should be bcp47 ==> expected: <true> but was: <false>
        org.opentest4j.AssertionFailedError: und-US-u-rg-ustx-tz-uschi: Should be bcp47 ==> expected: <true> but was: <false>
     …
```